xtext-gradle-plugin/src/main/java/org/xtext/gradle/XtextBuilderPlugin.xtend

```diff
@@ -86,9 +86,24 @@ class XtextBuilderPlugin implements Plugin<Project> {
 				return
 			}
 			if (requested.group == "org.eclipse.xtext" || requested.group == "org.eclipse.xtend") {
-				useVersion(version)
+				useVersion(version) // TODO
 			}
 		]
+		xtextTooling.resolutionStrategy.dependencySubstitution [
+			val version = xtextVersion.getVersion
+			if (version === null) {
+				return
+			}
+			
+			val ComparableVersion currentXtextVersion = new ComparableVersion(version);
+			val ComparableVersion targetVersion = new ComparableVersion("2.40.0")
+			if (currentXtextVersion >= targetVersion) {
```

Comment:
@oehme while we are parallel investigating with relocations on maven side, wdyt about this approach

Reply:
hmmm no idea why xtextVersion.getVersion crashes